### PR TITLE
Bw 757 - Unit test for leaderboard

### PR DESCRIPTION
**Pull request by milansar**

### CODE REVIEW

File: projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,307 @@
+import { async, ComponentFixture, TestBed, tick, fakeAsync } from '@angular/core/testing';
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts:1]: remove unused imports

File: projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+import { LeaderboardComponent } from './leaderboard.component';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { Store, MemoizedSelector } from '@ngrx/store';
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts:4]: remove unused import MemoizedSelector

File: projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+import { provideMockStore, MockStore } from '@ngrx/store/testing';
+import { Utils } from 'shared-library/core/services';
+import { User, Game, PlayerMode, GameStatus, Topic } from 'shared-library/shared/model';
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts:7]: remove unused imports

File: projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+                {provide: ActivatedRoute, useValue: {
+                      params: of(
+                        {
+                            category: 'id_query_params_test'
+                        }
+                      )
+                }},
+                TagActions,
+                TopicActions,
+                {provide: Utils, useValue: {
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts:40]: remove Utils as it is not used anywhere, remove related imports

File: projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('User dictionary should be empty initially', () => {
+          mockStore.refreshState();
+          expect(component.userDict).toEqual({});
+    });
+
+    it('logged In User should be empty intially', () => {
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts:89]: correct the spelling of intially

File: projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+            expect(component.loggedInUserId).toBeFalsy();
+    });
+
+    it('User should be empty intially', () => {
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts:93]: correct the spelling of initially

File: projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+        expect(component.user).toBeUndefined();
+    });
+
+    it('Leader board data be empty intially', () => {
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts:97]: correct spelling of initially

File: projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,307 @@
...
+        expect(component.leaderBoardCat).toEqual([]);
+    });
+
+    it('Leader board list start index intially should be falsy', () => {
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts:101]: correct spelling of initially,

File: projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,307 @@
...
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { Store, MemoizedSelector } from '@ngrx/store';
+import { provideMockStore, MockStore } from '@ngrx/store/testing';
+import { Utils } from 'shared-library/core/services';
+import { User, Game, PlayerMode, GameStatus, Topic } from 'shared-library/shared/model';
+import { AppState, appState } from '../../../store';
+import { TEST_DATA } from 'shared-library/testing/test.data';
+import { CoreState } from 'shared-library/core/store';
+import { MatSnackBarModule } from '@angular/material';
+import { UserActions, TagActions, TopicActions } from "shared-library/core/store/actions";
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts:12]: correct quote and other linting errors in file

File: projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,307 @@
...
+import { ActivatedRoute } from '@angular/router';
+import { of } from 'rxjs';
+import { DashboardState } from '../../store';
+
+describe('LeaderboardComponent', () => {
+
+    let component: LeaderboardComponent;
+    let fixture: ComponentFixture<LeaderboardComponent>;
+    let spy: any;
+    let user: User;
```
Comment (milansar) [inline projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts:22]: user is not used please remove it

File: projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts (hunk 4 above)
Comment (piya99) [reply projects/trivia/src/app/dashboard/component/leaderboard/leaderboard.component.spec.ts:40]: It is used to override Utils functions

**Comment by milansar** [comment]
@anihalaney , PR is ready for review

**Pull request merged**